plan/validator.go

```diff
@@ -456,6 +504,57 @@ func checkColumn(colDef *ast.ColumnDef) error {
 	return nil
 }
 
+func checkExistVariableExpr(expr ast.ExprNode) bool {
```

Comment:
What you do is visiting the ast tree. The correct way is to use `Visitor` pattern. See `validator` struct where you should do this check.